library/src/main/java/com/novoda/downloadmanager/WrappedOkHttpClient.java

```diff
@@ -0,0 +1,35 @@
+package com.novoda.downloadmanager;
+
+import com.squareup.okhttp.Call;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+
+import java.io.IOException;
+import java.util.Map;
+
+class WrappedOkHttpClient implements HttpClient {
+
+    private final OkHttpClient httpClient;
+
+    WrappedOkHttpClient(OkHttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public DownloadManagerResponse execute(DownloadManagerRequest request) throws IOException {
```

Comment:
This should make testing this class and the `FileDownloader` much easier 😄  `throws` because we still handles these in the calling code.